java/statestore/src/test/java/sleeper/statestore/transactionlog/TransactionLogStateStoreDynamoDBSpecificIT.java

```diff
@@ -216,43 +224,59 @@ void shouldLoadLatestFilesSnapshotIfNoPartitionsSnapshotIsPresent() throws Excep
                     .rootFirst("root")
                     .splitToNewChildren("root", "L", "R", 123L)
                     .buildTree();
-            FileReferenceFactory factory = FileReferenceFactory.from(tree);
+            FileReferenceFactory factory = factory(tree);
             List<FileReference> files = List.of(
                     factory.rootFile("file1.parquet", 100L),
                     factory.partitionFile("L", "file2.parquet", 25L),
                     factory.partitionFile("R", "file3.parquet", 50L));
-            saveFilesSnapshot(files, 2);
+            addFilesForSnapshot(files, 1);
+            createSnapshot();
 
             // When
-            StateStore stateStore = createStateStore();
+            StateStore stateStore = stateStore();
 
             // Then
             assertThat(stateStore.getAllPartitions()).isEmpty();
             assertThat(stateStore.getFileReferences())
-                    .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
                     .containsExactlyElementsOf(files);
         }
-    }
 
-    private void saveFilesSnapshot(List<FileReference> files, long transactionNumber) throws Exception {
-        StateStoreFiles state = new StateStoreFiles();
-        files.forEach(file -> state.add(fileWithOneReference(file, Instant.now())));
-        TransactionLogSnapshot snapshot = TransactionLogSnapshot.forFiles(tempDir.toString(), transactionNumber);
-        snapshotSerDe.saveFiles(snapshot, state);
-        snapshotStore.saveSnapshot(snapshot);
-    }
+        private void initialisePartitionsForSnapshot(PartitionTree tree, long transactionNumber) throws DuplicateTransactionNumberException {
+            partitionsStore.addTransaction(new TransactionLogEntry(transactionNumber, DEFAULT_UPDATE_TIME,
+                    new InitialisePartitionsTransaction(tree.getAllPartitions())));
+        }
+
+        private void addFilesForSnapshot(List<FileReference> files, long transactionNumber) throws DuplicateTransactionNumberException {
+            filesStore.addTransaction(new TransactionLogEntry(transactionNumber, DEFAULT_UPDATE_TIME,
+                    new AddFilesTransaction(files.stream()
+                            .map(file -> AllReferencesToAFile.fileWithOneReference(file, DEFAULT_UPDATE_TIME))
+                            .collect(Collectors.toList()))));
+        }
+
+        private void createSnapshot() {
+            new TransactionLogSnapshotCreator(
+                    instanceProperties, tableProperties, filesStore, partitionsStore, dynamoDBClient, configuration)
+                    .createSnapshot();
```

Comment:
If this was done inside saveFilesSnapshot/savePartitionsSnapshot as before, we could limit the scope of of the in-memory transaction log stores to just those methods. It's difficult to see what they're for.

We could potentially do this by taking a lambda to operate on a state store that will be backed by the in-memory logs. That would set up the state, and then when we pass the in-memory logs into the snapshot creator it would create the snapshots based on the same transactions.